ts/segment.go

```diff
@@ -40,12 +40,12 @@ type Segment struct {
 }
 
 // Len returns the length of the head and tail.
-func (s *Segment) Len() int {
+func (s Segment) Len() int {
 	return len(s.Head) + len(s.Tail)
 }
 
 // Finalize will release resources kept by the segment if pooled.
-func (s *Segment) Finalize() {
```

Comment:
Well golang has a magic shorthand where you just call `s.Finalize()` and it will essentially compile it as shorthand for `(&s).Finalize()`.  Hehe.